src/Microsoft.TemplateSearch.Common/Providers/NuGetMetadataSearchProvider.cs

```diff
@@ -70,15 +70,9 @@ internal class NuGetMetadataSearchProvider : ITemplateSearchProvider
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (_searchCache == null)
-            {
-                _logger.LogDebug("Initializing search cache...");
-                string metadataLocation = await GetSearchFileAsync(cancellationToken).ConfigureAwait(false);
-                _searchCache = TemplateSearchCache.FromJObject(_environmentSettings.Host.FileSystem.ReadObject(metadataLocation), _logger, _additionalDataReaders);
-                _logger.LogDebug("Search cache was successfully setup.");
-            }
+            await InitializeSearchCacheAsync(cancellationToken).ConfigureAwait(false);
 
-            IEnumerable<TemplatePackageSearchData> filteredPackages = _searchCache.TemplatePackages.Where(package => packFilter(package));
+            IEnumerable<TemplatePackageSearchData>? filteredPackages = _searchCache?.TemplatePackages.Where(package => packFilter(package));
```

Comment:
I change is because the IDE does not identify that we initialize `_searchCache` on the line before. Instead of `?` I'll change it to a `_searchCache!` to reflect better the assumed not null value at this point.

Reply:
is it change needed for original purpose of the PR?

Reply:
It is not, but the solution it does not build without this change.